ui-admin/src/study/participants/datarepo/DatasetList.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { getDatasetDashboardPath, StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+import Api, { DatasetDetails } from 'api/api'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { Store } from 'react-notifications-component'
+import { failureNotification } from 'util/notifications'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faExternalLink } from '@fortawesome/free-solid-svg-icons/faExternalLink'
+import { instantToDefaultString } from '../../../util/timeUtils'
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable
+} from '@tanstack/react-table'
+import { sortableTableHeader } from '../../../util/tableUtils'
+import { Link } from 'react-router-dom'
+import { faPlus } from '@fortawesome/free-solid-svg-icons/faPlus'
+import { useUser } from '../../../user/UserProvider'
+import CreateDatasetModal from './CreateDatasetModal'
+
+const DatasetList = ({ studyEnvContext }: {studyEnvContext: StudyEnvContextT}) => {
+  const { currentEnvPath } = studyEnvContext
+  const [showCreateDatasetModal, setShowCreateDatasetModal] = useState(false)
+  const [datasets, setDatasets] = useState<DatasetDetails[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [datasetsSorting, setDatasetsSorting] = React.useState<SortingState>([])
+  const { user } = useUser()
+
+  const datasetColumns = useMemo<ColumnDef<DatasetDetails, string>[]>(() => {
```

Comment:
same comment about pulling the column defs out

Reply:
Done in c30b794f7fa660fc64a29ad06b1e3583583f76a1